### PR TITLE
Moving RepairBrokenEolCharactersIfAny from SSH utils to common utils

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by robertschaedler3 — APPROVED** [review]

File: src/common/tests/CommonUtilsUT.cpp [hunk 1]
```diff
@@ -1664,4 +1664,28 @@ TEST_F(CommonUtilsTest, CheckLockoutForFailedPasswordAttempts)
...
+    EXPECT_STREQ(expected, value = CleanDoubleBackslashes("\nThis is a test\n\nHere is another line\nAnd another\n\n\nEnd\n"));
+    FREE_MEMORY(value);
+
+    EXPECT_STREQ(expected, value = CleanDoubleBackslashes("\\nThis is a test\n\nHere is another line\nAnd another\\n\n\\nEnd\\n"));
+    FREE_MEMORY(value);
+
+    EXPECT_STREQ(expected, value = CleanDoubleBackslashes("\nThis is a test\\n\\nHere is another line\\nAnd another\\n\n\\nEnd\n"));
+    FREE_MEMORY(value);
+
+    EXPECT_STREQ("\n\\Test\\123\n", value = CleanDoubleBackslashes("\n\\Test\\123\\n"));
```
Comment (AhmedBM) [inline src/common/tests/CommonUtilsUT.cpp:1689]: Only seeing single backslashes `\` and not double `\\` in your tests, should it also support double, as your function name suggests?

```
\\ This is a test \\
\\ Here is another line \\
\\ And another \\
```

Comment (MariusNi) [reply src/common/tests/CommonUtilsUT.cpp:1689]: The double backslash this function addresses is "\n" being changed to "\\n" and with this EOL getting broken intro 2 meaningless characters. I'll add a comment to explain this for the function.

Comment (AhmedBM) [reply src/common/tests/CommonUtilsUT.cpp:1689]: ahh ok i see! Thanks for the explanation! Comments explaining this would be great 👍

Comment (MariusNi) [reply src/common/tests/CommonUtilsUT.cpp]: Better, changed the function name.

**Review by AhmedBM — APPROVED** [review]
👌

**Review by AhmedBM — APPROVED** [review]
👌

**Pull request merged**